crates/polars-core/src/chunked_array/builder/list/mod.rs

```diff
@@ -130,6 +130,13 @@ pub fn get_list_builder(
             list_capacity,
             Some(inner_type_logical.clone()),
         ))),
+        #[cfg(feature = "dtype-decimal")]
+        DataType::Decimal(_, _) => Ok(Box::new(ListPrimitiveChunkedBuilder::<Int128Type>::new(
+            name,
+            list_capacity,
+            value_capacity,
+            inner_type_logical.clone(),
+        ))),
```

Comment:
I would have simply expanded the `match_dtype_to_logical_apply_macro` but there is no `Int128` dtype, so instead we have to directly supply the type here. Do we want to keep `Int128` hidden or should we make a new dtype?

Reply:
It should still remain hidden. Maybe we opt for supporting it one day, but not now.